### PR TITLE
Run Mac hostonly tests on any available arch

### PR DESCRIPTION
**Pull request by jmagman**
M1 capacity has now been added.  Another attempt at https://github.com/flutter/flutter/pull/109889.

Allow Mac hostonly (not devicelab or benchmark) tests to run on any available architecture, not just x64 Intel machines.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -78,7 +78,7 @@ platform_properties:
         ]
       os: Mac-12
       device_type: none
-      cpu: x86 # TODO(jmagman): https://github.com/flutter/flutter/issues/112130
+      cpu: arm64
```
Comment (jmagman) [inline .ci.yaml:81]: This line will be removed before merge.  I've temporarily set this to arm to force all "Mac" tests to run there to validate they pass on this PR.

File: .ci.yaml [hunk 2]
```diff
@@ -4914,8 +4904,6 @@ targets:
       task_name: flutter_packaging
       tags: >
         ["framework", "hostonly", "shard", "mac"]
-    dimensions:
-      cpu: "arm64"
```
Comment (jmagman) [inline .ci.yaml:4918]: I believe this is redundant since `Mac_arm64` is specified.

**Comment by jmagman** [comment]
`Mac plugin_test_ios` timed out at 30 minutes, even though it's only taking 14 minutes in prod.  Will keep that one on x64 while it's being investigated: https://github.com/flutter/flutter/issues/119764

I also can't move the web tests onto arm due to a Chrome cipd tag issue: https://github.com/flutter/flutter/issues/110113#issuecomment-1412885655

File: .ci.yaml [hunk 3]
```diff
@@ -2697,8 +2696,6 @@ targets:
...
```
Comment (jmagman) [inline .ci.yaml:2701]: I believe this is redundant since `Mac_arm64` is specified.

File: .ci.yaml [hunk 4]
```diff
@@ -3067,7 +3065,6 @@ targets:
     recipe: flutter/flutter_drone
     timeout: 60
     properties:
-      cpu: arm64
```
Comment (jmagman) [inline .ci.yaml:3070]: Same.

**Comment by jmagman** [comment]
Reverted the changes to force "Mac" tests to run on arm, as well as the web tests that bumped Chrome version (will address that in another PR with input from the web team).

**Review by yusuf-goog — APPROVED** [review]

**Comment by jmagman** [comment]
This caused the hostonly Mac benchmarks to run on bots with different specs: https://github.com/flutter/flutter/issues/119780

**Pull request merged**